hw/ip/otbn/dv/otbnsim/sim/sim.py

```diff
@@ -44,29 +44,28 @@ def step(self, verbose: bool) -> Tuple[Optional[OTBNInsn], List[Trace]]:
         if not self.state.running:
             return (None, [])
 
-        was_stalled = self.state.stalled
         pc_before = self.state.pc
 
-        if was_stalled:
+        word_pc = int(self.state.pc) >> 2
+        if word_pc >= len(self.program):
+            raise RuntimeError('Trying to execute instruction at address '
+                               '{:#x}, but the program is only {:#x} '
+                               'bytes ({} instructions) long. Since there '
+                               'are no architectural contents of the '
+                               'memory here, we have to stop.'
+                               .format(int(self.state.pc),
+                                       4 * len(self.program),
+                                       len(self.program)))
+        insn = self.program[word_pc]
+
+        sim_stalled = (self.state.non_insn_stall or
+                       not insn.pre_execute(self.state))
+
+        if sim_stalled:
+            self.state.commit(sim_stalled=True)
             insn = None
```

Comment:
Now that we define `insn` outside of the if/else block, can we drop this (so `insn` can't be `None`) and move the definition of disasm into the if/else block?